File: java/src/main/java/sammancoaching/LeadershipTeam.java
package sammancoaching;

class LeadershipTeam implements SeniorityLevel {
    @Override
    public double getPensionContributionBonus(SalaryContributionPercentages databaseAccess) {
        // BUG: Should be LEADERSHIP_TEAM_PERCENTAGE
        return databaseAccess.lookupValue(SalaryContributionPercentages.MID_SENIORITY_PERCENTAGE);
    }
}

File: typescript/src/LeadershipTeam.ts
import {ContributionPercentages, SalaryContributionPercentages} from "./SalaryContributionPercentages";
import {SeniorityLevel} from "./SeniorityLevel";

export class LeadershipTeam implements SeniorityLevel {
  getPensionContributionBonus(databaseAccess: ContributionPercentages): number {
    // BUG: Should be LEADERSHIP_TEAM_PERCENTAGE
    return databaseAccess.lookupValue(SalaryContributionPercentages.MID_SENIORITY_PERCENTAGE);
  }
}
